git-hub

```diff
@@ -1464,6 +1464,19 @@ class PullUtil (IssueUtil):
         infof('Fetching {} from {}', ref, url)
         git_quiet(1, 'fetch', '--', url, ref)
 
+    # Checks if the base branch exists (and die if it doesn't)
+    @staticmethod
+    def ensure_github_branch_exists(base):
+        try:
+            req.get('/repos/{}/branches/{}'.format(config.upstream, base))
```

Comment:
String interpolation might be a possibility here unless you intend to support Python 3.5 and earlier.

Reply:
Right. I wasn't specifically thinking about compatibility (it was just an oversight/forgot about `f''`) but since we come from Python2, I guess not jumping too far to the future might be desirable. :P (I think this will be the only thing making it incompatible with old Python3 versions).

Reply:
I'll merge it as is, as there are many old idioms still present, I prefer not to decide at this stage which should be the oldest supported Python release.